web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/UserServiceImpl.java

```diff
@@ -89,4 +103,31 @@ private String getRole(WebCertUser webCertUser) {
         return webCertUser.getRoleTypeName().equalsIgnoreCase("VARDADMINISTRATOR") ? "Vårdadministratör"
             : webCertUser.getRoleTypeName().split(" ")[0];
     }
+
+    private Vardenhet getLoggedInCareUnit(WebCertUser webCertUser) {
+        final var loggedInCareProviderId = webCertUser.getValdVardgivare().getId();
+        final var loggedInUnitId = webCertUser.getValdVardenhet().getId();
+        final var currentCareProvider = webCertUser.getVardgivare().stream()
+            .filter(careProvider -> careProvider.getId().equals(loggedInCareProviderId)).findFirst().orElse(null);
```

Comment:
As the code is written, it can leave the care unit as null. But if the careunit is null, then the state of the user-object is invalid. 

Probably better to throw an exception instead of returning null, which will remove the need of many nullchecks.